subsys/usb/Kconfig

```diff
@@ -36,6 +36,19 @@ config SYS_LOG_USB_DEVICE_LEVEL
 
 	  - 4 DEBUG, write SYS_LOG_DBG in addition to previous levels
 
+config USB_SPEC_VERSION
```

Comment:
This change needs more discussion. It gets kind of messed up here USB2.0 spec and high speed part of it. I briefly looked over the USB controller drivers, as I see they are all USB2.0 full speed part compliant. Whether the controller also supports high speed or not can/should be configured via DT. The device stack itself only supports full speed and needs to be adapted accordingly.

Reply:
What changes do you think the device stack needs to support high speed? I tested these changes on an Intel S1000 that supports high speed and atleast for the HID class that I tested, there were no changes needed in the device stack.

Reply:
I hope I do not mess it up. High-speed capable device must support other speed requests, see "9.2.6.6 Speed Dependent Descriptors". The host can request it, but does not have to. And I do not see any objection to raising bcdUSB to 2.00 in general. We just have to make sure that only High-speed capable devices respond to device_qualifier (9.6.2.) descriptor and other_speed_configuration requests. @finikorg your opinion?

Reply:
Thanks @jfischer-phytec-iot for the pointer to the sections in the USB 2.0 spec.
Reading the specification wording, it also looks like speed dependent configurations need not always be present unless the device does have a different configuration for different speeds.
>However, a high-speed capable device **may** have configurations that are speed dependent.

I'd like to see @finikorg opinion as well.

Reply:
I think it is OK to raise  `bcdUSB` to 2.00. (We need this also for MS OS Descriptors).

Reply:
@jfischer-phytec-iot I could drop the commit, however the commit also has
* `MAX_HIGH_POWER` vs `MAX_LOW_POWER` selection
* Max packet size for HID INT EP
How should these be handled?

Reply:
- bMaxPower, it was not nice, and this commit does not make it any better, I would leave it as it was and open an issue.

- MPS of 16 is valid size for high-speed interrupt endpoints (0 to 1024 bytes), you can add `prompt` to HID_INTERRUPT_EP_MPS and make it configurable, then it is possible to configure MPS in your boards or app configuration file. Please if you want to do it, then in a different PR/commit.

Reply:
commit is now dropped.